adal/src/main/java/com/microsoft/aad/adal/AuthenticationException.java

```diff
@@ -33,6 +41,12 @@
 
     private ADALError mCode;
 
+    private Map<String, String> mResponseBody;
+
+    private int mServiceStatusCode = -1;
+
+    private Map<String, List<String>> mHttpResponseHeaders;
```

Comment:
not the right data structure, the spec says Map<String, String>, use HashMapExtensions.getJsonResponse

Reply:
Hi @addev-ashish The spec didn't say the header is in type Map<String, String>. In addition, the responseHeaders in ADAL is defined as Map<String, List<String>>, please refer to HttpURLConnection.getHeaderFields().

Reply:
https://developer.android.com/reference/java/net/URLConnection.html#getHeaderFields()